gn/build/config/compiler/BUILD.gn

```diff
@@ -58,7 +58,19 @@ config("abi_default") {
     if (arm_float_abi != "") {
       cflags += [ "-mfloat-abi=${arm_float_abi}" ]
     }
-    if (arm_use_thumb) {
+    if (is_clang) {
+      if (current_os == "ios" || current_os == "mac") {
+        cflags += [ "-target",
+          string_join("-",
+              [
+                getenv("ARCHS"),
+                getenv("LLVM_TARGET_TRIPLE_VENDOR"),
+                getenv("LLVM_TARGET_TRIPLE_OS_VERSION"),
+          ])
+        ]
+      }
+    }
```

Comment:
@mspang would really appreciate you looking to see if this can be accomplished another/better way, e.g. another parameter in BUILDCONFIG.gn?

Reply:
Is the OS version the gap? I'd assume you can set vendor based on current_os and architecture based on current_cpu (plus details from an architecture specific build import e.g. arm.gni). For OS versions we could add a mac_sdk.gni and conditionally import it into this file. Chromium has one, https://source.chromium.org/chromium/chromium/src/+/master:build/config/mac/mac_sdk.gni . It looks quite a bit more complex but might be useful to look at.

I do think we should avoid getenv() as much as possible as it results in correctness issues. ninja won't regen the build files based on environment changes, it only regenerates after updates to out/args.gn and other build files.

Reply:
I have moved this to the connector